cli/standalone.rs

```diff
@@ -1,45 +1,111 @@
 use deno_core::error::AnyError;
 use deno_core::error::Context;
 use deno_core::futures::FutureExt;
+use deno_core::serde::de;
 use deno_core::serde::Deserialize;
+use deno_core::serde::Deserializer;
 use deno_core::serde::Serialize;
+use deno_core::serde::Serializer;
 use deno_core::serde_json;
 use deno_core::v8_set_flags;
 use deno_core::ModuleLoader;
 use deno_core::ModuleSpecifier;
 use deno_core::OpState;
 use deno_runtime::permissions::Permissions;
+use deno_runtime::permissions::PermissionsOptions;
 use deno_runtime::worker::MainWorker;
 use deno_runtime::worker::WorkerOptions;
+use log::Level;
 use std::cell::RefCell;
 use std::convert::TryInto;
 use std::env::current_exe;
-use std::fs::read;
+use std::fmt;
 use std::fs::File;
 use std::io::Read;
 use std::io::Seek;
 use std::io::SeekFrom;
-use std::io::Write;
 use std::iter::once;
-use std::path::PathBuf;
 use std::pin::Pin;
 use std::rc::Rc;
+use std::str::FromStr;
 use std::sync::Arc;
 
 #[derive(Deserialize, Serialize)]
-struct Metadata {
-  flags: Flags,
-  ca_data: Option<Vec<u8>>,
+pub struct Metadata {
+  pub argv: Vec<String>,
+  pub unstable: bool,
+  pub seed: Option<u64>,
+  pub permissions: PermissionsOptions,
+  pub v8_flags: Vec<String>,
+  #[serde(deserialize_with = "deserialize_maybe_log_level")]
+  #[serde(serialize_with = "serialize_maybe_log_level")]
+  pub log_level: Option<Level>,
+  pub ca_data: Option<Vec<u8>>,
 }
 
-const MAGIC_TRAILER: &[u8; 8] = b"d3n0l4nd";
+fn deserialize_maybe_log_level<'de, D>(d: D) -> Result<Option<Level>, D::Error>
+where
+  D: Deserializer<'de>,
+{
+  struct OptionalLogLevelVisitor;
+  impl<'de> de::Visitor<'de> for OptionalLogLevelVisitor {
+    type Value = Option<Level>;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+      write!(formatter, "null or a valid log level string")
+    }
+
+    fn visit_none<E>(self) -> Result<Self::Value, E>
+    where
+      E: de::Error,
+    {
+      Ok(None)
+    }
+
+    fn visit_some<D>(self, d: D) -> Result<Self::Value, D::Error>
+    where
+      D: de::Deserializer<'de>,
+    {
+      struct LogLevelVisitor;
+      impl<'de> de::Visitor<'de> for LogLevelVisitor {
+        type Value = Level;
+
+        fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+          write!(formatter, "a valid log level string")
+        }
+
+        fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+        where
+          E: de::Error,
+        {
+          Level::from_str(s).map_err(|_| {
+            de::Error::invalid_value(de::Unexpected::Str(s), &self)
+          })
+        }
+      }
+      Ok(Some(d.deserialize_str(LogLevelVisitor)?))
+    }
+  }
+  d.deserialize_option(OptionalLogLevelVisitor)
+}
+
+fn serialize_maybe_log_level<S>(
+  maybe_level: &Option<Level>,
+  s: S,
+) -> Result<S::Ok, S::Error>
+where
+  S: Serializer,
+{
+  match maybe_level {
+    None => s.serialize_none(),
+    Some(level) => s.serialize_str(&level.to_string()),
+  }
+}
```

Comment:
This seems like a lot of code to serialize the log level...

Reply:
Unfortunately... I already expressed it in https://github.com/denoland/deno/pull/8738#discussion_r551052751

Reply:
Actually, just needed to activate "serde" feature in "log" crate